File: 435-non-overlapping-intervals/435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        //sort by the end
        int res = 0;
        Arrays.sort(intervals, (a, b) -> a[1]- b[1]);
        int end = Integer.MIN_VALUE;
        for(int[] interval : intervals) {
            if(interval[0] < end) {
                res++;
            } else {
                end = interval[1];
            }
        }
        return res;
    }
}

File: 435-non-overlapping-intervals/435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x : (x[1], -x[0]))
        prev = intervals[0]
        cnt = 0
        for i in intervals[1:]:
            if i[0] < prev[1]: #overlap
                cnt += 1
            else:
                prev = i
        return cnt
            
        